aws_crt_common_ws/aws-crt-common-sys/build.rs

```diff
@@ -0,0 +1,337 @@
+        .is_ok()
+    {
+        build_config.add_public_define("AWS_HAVE_WINAPI_DESKTOP", "1");
+        config_file_str = format!("{}#define AWS_HAVE_WINAPI_DESKTOP 1\n", config_file_str);
+    }
+
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    {
+        build_config.add_public_define("AWS_ARCH_INTEL", "1");
+        config_file_str = format!("{}#define AWS_ARCH_INTEL 1\n", config_file_str);
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    {
+        build_config.add_public_define("AWS_ARCH_ARM64", "1");
+        config_file_str = format!("{}#define AWS_ARCH_ARM64 1\n", config_file_str);
+    }
+
+    #[cfg(target_arch = "arm")]
+    {
+        build_config.add_public_define("AWS_ARCH_ARM32", "1");
+        config_file_str = format!("{}#define AWS_ARCH_ARM32 1\n", config_file_str);
+    }
+
+    if build_config
+        .try_compile(
+            "int main() {
+        int foo = 42, bar = 24;
+        __asm__ __volatile__(\"\":\"=r\"(foo):\"r\"(bar):\"memory\");\
+        return 0;\
+        }",
+        )
+        .is_ok()
+    {
+        build_config.add_private_define("AWS_HAVE_GCC_INLINE_ASM", "1");
+        config_file_str = format!("{}#define AWS_HAVE_GCC_INLINE_ASM 1\n", config_file_str);
+    }
+
+    if build_config
+        .try_compile(
+            "#include <sys/auxv.h>
+        int main() {
+        #ifdef __linux__
+            getauxval(AT_HWCAP);
+            getauxval(AT_HWCAP2);
+        #endif
+        return 0;
+        }",
+        )
+        .is_ok()
+    {
+        build_config.add_private_define("AWS_HAVE_AUXV", "1");
+        config_file_str = format!("{}#define AWS_HAVE_AUXV 1\n", config_file_str);
+    }
+
+    if build_config
+        .try_compile(
+            "#include <execinfo.h>
+        int main() {
+            return 0;
+        }",
+        )
+        .is_ok()
+    {
+        build_config.add_private_define("AWS_HAVE_EXECINFO", "1");
+        config_file_str = format!("{}#define AWS_HAVE_EXECINFO 1\n", config_file_str);
+    }
+
+    if build_config
+        .try_compile(
+            "#include <unistd.h>
+        int main() {
+            sysconf(_SC_NPROCESSORS_ONLN);
+            return 0;
+        }",
+        )
+        .is_ok()
+    {
+        build_config.add_private_define("HAVE_SYSCONF", "1");
+        config_file_str = format!("{}#define HAVE_SYSCONF 1\n", config_file_str);
+    }
+    config_file_str = format!("{}#endif\n", config_file_str);
+    let output_path = Path::new("include/aws/common/config.h");
+    build_config
+        .write_generated_file_to_output_path(config_file_str.to_string().borrow(), output_path);
+}
+fn main() {
+    let mut build_info = CRTModuleBuildInfo::new("aws-c-common");
+    load_compiler_configuration(build_info.borrow_mut());
+    let include_path = Path::new("../../include/aws");
+    build_info.add_include_dir_and_copy_to_build_tree(include_path);
+
+    build_info
+        .add_file_to_build(Path::new("../../source/allocator.c"))
```

Comment:
is there every file in `source`? Might want https://docs.rs/include_dir/0.6.0/include_dir/

Reply:
it's not every file in source and the files are often platform specific, so in this case I'd prefer to keep it explicit.